### PR TITLE
Alma documentation cleanup

### PR DESCRIPTION
**Pull request by tinumide**

### CODE REVIEW

File: CHANGES.rst
Comment (bsipocz) [inline CHANGES.rst:11]: no changelog entry is needed for doc changes

File: docs/alma/alma.rst
Comment (bsipocz) [inline docs/alma/alma.rst:128]: interestingly I run into an error here. It's certainly unrelated to the PR, but would hold back merge until it's resolved.

**Review by bsipocz — COMMENTED** [review]
There are unrelated failures, ideally those should be resolved before merging this, to ensure the cleaned up docs indeed passes tests.

File: docs/alma/alma.rst
Comment (bsipocz) [inline docs/alma/alma.rst:130]: hmm, right now I get 400 for this, so I think we should probably drop this print altogether. If nothing has returned then the tests will fail with the next line anyway

File: docs/alma/alma.rst
Comment (bsipocz) [reply docs/alma/alma.rst:128]: When the tests are not run together with the rest of the module's test suite, then it's more sensible, and relevant failures are found. So let's keep running `pytest docs/alma/alma.rst --remote-data` for this PR.

**Comment by bsipocz** [comment]
@keflavich - could you take over this PR to enable doctesting on the alma docs?

**Comment by keflavich** [comment]
I'm working on it, but I get this weird error:
```
astropy.utils.exceptions.AstropyWarning: leap-second auto-update failed due to the following exception: DeprecationWarning('`np.int` is a deprecated alias for the builtin `int`. To silence this warning, use `int` by itself. Doing this will not modify any behavior and is safe. When replacing `np.int`, you may wish to use e.g. `np.int64` or `np.int32` to specify the precision. If you wish to review your current use, check the release note link for additional information.\nDeprecated in NumPy 1.20; for more details and guidance: https://numpy.org/devdocs/release/1.20.0-notes.html#deprecations')
/Users/adam/repos/astroquery/docs/alma/alma.rst:25: UnexpectedException
```
that is unrelated.  This feels like something caused by an out-of-date astropy, but I'm testing w/`5.1.dev184+gbf35e6f60b`

**Comment by bsipocz** [comment]
There are two unrelated commits picked up. I'll look into how we could filter out that leap second warning, and also ping @eteq as this leap second thing comes up regularly, so there may be a solution for it that I cannot remember from the top of my head.

**Comment by bsipocz** [comment]
@keflavich - That leap-second warning is already on the ignore list, so I'm not sure why it's causing issues locally. Did you see it before or after you made the rebase?

**Comment by keflavich** [comment]
I fixed the rebase - some weird things happened that I didn't understand.   I had to rebase several times, even after updating the remote.

I had to add a different warning catch for this warning:
```
DeprecationWarning: `np.int` is a deprecated alias for the builtin `int`. To silence this warning, use `int` by itself. Doing this will not modify any behavior and is safe. When replacing `np.int`, you may wish to use e.g. `np.int64` or `np.int32` to specify the precision. If you wish to review your current use, check the release note link for additional information.
Deprecated in NumPy 1.20; for more details and guidance: https://numpy.org/devdocs/release/1.20.0-notes.html#deprecations
/Users/adam/repos/astroquery/docs/alma/alma.rst:25: UnexpectedException
```

but I don't think that is right, something local must be broken again.

File: docs/alma/alma.rst
Comment (bsipocz) [inline docs/alma/alma.rst:129]: this line still gives me a failure (it returns with the original line that you've changed), I would suggest adding `# doctest: +IGNORE_OUTPUT` to the end of this line, so it tests that it runs without issue, but doesn't choke on if it changes. The actual column names then can be tested as part of the test suite, if needed.

Comment (keflavich) [reply docs/alma/alma.rst]: weird that it's changing order, but I should have been using `IGNORE_OUTPUT` in several places.

File: setup.cfg
Comment (bsipocz) [inline setup.cfg:42]: We shouldn't do this as we need to catch deprecations that we trigger with astroquery code.

Comment (keflavich) [reply setup.cfg:42]: ach, yes, this was a mistake: for reasons I don't understand, I have to have this for tests to pass locally, but I didn't mean to commit it.

**Comment by keflavich** [comment]
I'm getting a ton of exceptions like these, but they're totally unrelated to this PR.  I think they're slowing down/screwing with the testing.

```
Exception ignored in: <ssl.SSLSocket fd=21, family=AddressFamily.AF_INET, type=SocketKind.SOCK_STREAM, proto=6, laddr=('10.13.167.149', 39336), raddr=('192.33.115.31', 443)>
Traceback (most recent call last):
  File "/orange/adamginsburg/miniconda3/envs/python39/lib/python3.9/site-packages/_pytest/unraisableexception.py", line 59, in __exit__
    del self.unraisable
ResourceWarning: unclosed <ssl.SSLSocket fd=21, family=AddressFamily.AF_INET, type=SocketKind.SOCK_STREAM, proto=6, laddr=('10.13.167.149', 39336), raddr=('192.33.115.31', 443)>
```

**Comment by keflavich** [comment]
I also see tons of these:
```
Exception ignored in: <ssl.SSLSocket fd=21, family=AddressFamily.AF_INET, type=SocketKind.SOCK_STREAM, proto=0, laddr=('10.0.0.44', 57697), raddr=('192.33.115.31', 443)>
ResourceWarning: unclosed <ssl.SSLSocket fd=21, family=AddressFamily.AF_INET, type=SocketKind.SOCK_STREAM, proto=0, laddr=('10.0.0.44', 57697), raddr=('192.33.115.31', 443)>
```
but... why?  We shouldn't have unclosed sockets, right?  We're just doing a few requests?  Maybe this is within `requests`?

**Comment by bsipocz** [comment]
I recall seeing those socket ones occasionally, but don't think it's related to the content of this PR, and suspect they are upstream either pytest or request. Do you recall seeing any from the python terminal, or they only pop up while you run the tests?

**Comment by keflavich** [comment]
I don't see them on the command line, just during tests.  

The doctests are super slow to run; I think they're downloading a lot of data.  That might be necessary, but it certainly has made testing inconvenient.

**Comment by bsipocz** [comment]
@keflavich - I also see that they run a very long time, though I don't have many files permanently downloaded. It would be good if the problematic ones would be rephrased, or at least skipped. I would prefer the rewrite as I think it's more useful for users if they can copy-paste examples that actually run and finishes within a reasonable time, they can then rescale it for their own use case.

**Comment by keflavich** [comment]
yes I fully agree.  I'll try to work through them again and see if I can pinpoint the misbehaving example(s) and thin them out.  Some of the issue is likely that there is much more data in the archive now than when the examples were written.

**Comment by bsipocz** [comment]
> Some of the issue is likely that there is much more data in the archive now than when the examples were written.

I agree that's likely the case, we run into the same issue with mast not long ago.

**Comment by keflavich** [comment]
I'm finding this PR extremely difficult to test because the service keeps timing out.  It's not that clear which sorts of queries are actually fast; maybe the archive is doing a lot of work when we ask for a simple README?

File: docs/alma/alma.rst
Comment (keflavich) [reply docs/alma/alma.rst:128]: this query works fine for me right now and produces 400 lines of talbe, so I consider this resolved.

(but yes let's keep running the tests...)

File: docs/alma/alma.rst
Comment (keflavich) [inline docs/alma/alma.rst:185]: This example times out for me, so I tried a simpler one and got an error:
https://github.com/astropy/astroquery/issues/2240

File: docs/alma/alma.rst
Comment (keflavich) [inline docs/alma/alma.rst:177]: this version didn't work; it appears to have returned _only_ non-public data, and there is no longer any non-public data

File: docs/alma/alma.rst
Comment (keflavich) [inline docs/alma/alma.rst:341]: the "download a tarball, extract, delete" approach was an old hack; this is the _right_ way to do this.

**Comment by keflavich** [comment]
I'm afraid this is as good as it's going to get for now.  There are several queries that routinely time out during tests, but I think that's a server issue.  I've tried finding smaller items to query for, but that apparently only goes so far.

I suggest we merge - assuming there are no flagrant errors - and try to revisit this as other issues are resolved.

**Comment by bsipocz** [comment]
OK, sounds like a plan. If it's always the same examples that time out, I would err on the side of keep those ones skipped.

And maybe also open a feature request for doctestplus/remote-data that adds a timeout option for testing? How would you @pllim or @saimn  feel about that? Getting a failed status in an upper bound duration I feel is more valuable that having the whole test process stalled due to some server issues.

**Comment by keflavich** [comment]
I like the idea of putting limits on the timeout.

The NRAO servers went down today for a major overhaul (as I found out by getting 100% timeouts this morning...), so maybe they'll behave better when they come back up...

**Comment by bsipocz** [comment]
Oh, wait a minute. Wouldn't the timeout in our own config system solve this? Somehow I don't feel the 60s is honoured, as the test for that narrative docs file runs forever.

**Comment by pllim** [comment]
Re: timeout -- Depends on what is actually getting called for a particular remote data call. Remote data directive simply allows it to run, but the actual timeout limit is enforced by whatever is getting called. Do you have a specific example? I haven't been following this repo closely so I don't know what code is being affected here. Thanks!

File: docs/alma/alma.rst
Comment (keflavich) [reply docs/alma/alma.rst:177]: should be `public=None`

**Comment by keflavich** [comment]
I think we should go ahead and merge this, but we might need a fresh PR to clean up some more details on another (future) review.

Has the readthedocs build been pending since christmas?

**Comment by pllim** [comment]
There is a problem with your build: https://readthedocs.org/projects/astroquery/builds/15892183/

**Comment by bsipocz** [comment]
@keflavich - does this build locally? As I recall it was hung up for me when I tried it before the holidays. We cannot merge if it hangs as it would block the rest of the package.

**Comment by keflavich** [comment]
I didn't realize there was a real build issue here.  I'll investigate when I get back to this.

**Comment by bsipocz** [comment]
This PR should also remove or refactor the test `test_doc_example` in the remote tests file, too.

File: docs/alma/alma.rst
Comment (bsipocz) [inline docs/alma/alma.rst:129]: This should be IGNORE_OUTPUT rather than skip
```suggestion
    >>> m83_data.colnames  # doctest: +IGNORE_OUTPUT
```

File: docs/alma/alma.rst
Comment (bsipocz) [inline docs/alma/alma.rst:297]: Are these expected to change? If not, then we shall  not ignore

File: docs/alma/alma.rst
Comment (bsipocz) [inline docs/alma/alma.rst:329]: as all is skipped here, maybe leave it without remote data

**Review by bsipocz — COMMENTED** [review]
I'll try to rebase this and will squash a few commits to address my comments. Hopefully that will trigger RDT and it won't timeout on us any longer.

**Review by bsipocz — APPROVED** [review]
Passing locally, so we're good to go if the RTD rendering works.

**Comment by bsipocz** [comment]
Hmm, tests pass, but sphinx is still not happy, atm it's an IndexError: https://readthedocs.org/projects/astroquery/builds/16411376/

```
reading sources... [  0%] alma/alma
/home/docs/checkouts/readthedocs.org/user_builds/astroquery/envs/1935/lib/python3.8/site-packages/astroquery/ipac/irsa/sha/__init__.py:14: UserWarning: Experimental: SHA has not yet been refactored to have its API match the rest of astroquery.
  warnings.warn("Experimental: SHA has not yet been refactored to have its "
/home/docs/checkouts/readthedocs.org/user_builds/astroquery/envs/1935/lib/python3.8/site-packages/astroquery/alfalfa/__init__.py:15: UserWarning: Experimental: ALFALFA has not yet been refactored to have its API match the rest of astroquery.
  warnings.warn("Experimental: ALFALFA has not yet been refactored to have "
WARNING: Astrometry.net API key not found in configuration file [astroquery.astrometry_net.core]
WARNING: You need to manually edit the configuration file and add it [astroquery.astrometry_net.core]
WARNING: You may also register it for this session with AstrometryNet.key = 'XXXXXXXX' [astroquery.astrometry_net.core]
/home/docs/checkouts/readthedocs.org/user_builds/astroquery/envs/1935/lib/python3.8/site-packages/astroquery/ogle/__init__.py:40: UserWarning: Experimental: OGLE has not yet been refactored to have its API match the rest of astroquery.
  warnings.warn("Experimental: OGLE has not yet been refactored to have its "
WARNING: AstropyDeprecationWarning: the module relies on an unmaintained library and isconsidered deprecated until completely refactored or upstreamis stablised. [astroquery.vamdc.core]
/home/docs/checkouts/readthedocs.org/user_builds/astroquery/envs/1935/lib/python3.8/site-packages/astroquery/vamdc/core.py:116: UserWarning: vamdclib could not be imported; the vamdc astroquery module will not work
  warnings.warn("vamdclib could not be imported; the vamdc astroquery module "
WARNING: AstropyDeprecationWarning: the module relies on an unmaintained library and isconsidered deprecated until completely refactored or upstreamis stablised. [astroquery.vamdc.core]
/home/docs/checkouts/readthedocs.org/user_builds/astroquery/envs/1935/lib/python3.8/site-packages/astroquery/fermi/__init__.py:37: UserWarning: Experimental: Fermi-LAT has not yet been refactored to have its API match the rest of astroquery.
  warnings.warn("Experimental: Fermi-LAT has not yet been refactored to have "

Traceback (most recent call last):
  File "/home/docs/checkouts/readthedocs.org/user_builds/astroquery/envs/1935/lib/python3.8/site-packages/sphinx/cmd/build.py", line 284, in build_main
    app.build(args.force_all, filenames)
  File "/home/docs/checkouts/readthedocs.org/user_builds/astroquery/envs/1935/lib/python3.8/site-packages/sphinx/application.py", line 337, in build
    self.builder.build_update()
  File "/home/docs/checkouts/readthedocs.org/user_builds/astroquery/envs/1935/lib/python3.8/site-packages/sphinx/builders/__init__.py", line 294, in build_update
    self.build(to_build,
  File "/home/docs/checkouts/readthedocs.org/user_builds/astroquery/envs/1935/lib/python3.8/site-packages/sphinx/builders/__init__.py", line 308, in build
    updated_docnames = set(self.read())
  File "/home/docs/checkouts/readthedocs.org/user_builds/astroquery/envs/1935/lib/python3.8/site-packages/sphinx/builders/__init__.py", line 415, in read
    self._read_serial(docnames)
  File "/home/docs/checkouts/readthedocs.org/user_builds/astroquery/envs/1935/lib/python3.8/site-packages/sphinx/builders/__init__.py", line 436, in _read_serial
    self.read_doc(docname)
  File "/home/docs/checkouts/readthedocs.org/user_builds/astroquery/envs/1935/lib/python3.8/site-packages/sphinx/builders/__init__.py", line 476, in read_doc
    doctree = read_doc(self.app, self.env, self.env.doc2path(docname))
  File "/home/docs/checkouts/readthedocs.org/user_builds/astroquery/envs/1935/lib/python3.8/site-packages/sphinx/io.py", line 189, in read_doc
    pub.publish()
  File "/home/docs/checkouts/readthedocs.org/user_builds/astroquery/envs/1935/lib/python3.8/site-packages/docutils/core.py", line 217, in publish
    self.document = self.reader.read(self.source, self.parser,
  File "/home/docs/checkouts/readthedocs.org/user_builds/astroquery/envs/1935/lib/python3.8/site-packages/sphinx/io.py", line 109, in read
    self.parse()
  File "/home/docs/checkouts/readthedocs.org/user_builds/astroquery/envs/1935/lib/python3.8/site-packages/docutils/readers/__init__.py", line 78, in parse
    self.parser.parse(self.input, document)
  File "/home/docs/checkouts/readthedocs.org/user_builds/astroquery/envs/1935/lib/python3.8/site-packages/sphinx/parsers.py", line 101, in parse
    self.statemachine.run(inputlines, document, inliner=self.inliner)
  File "/home/docs/checkouts/readthedocs.org/user_builds/astroquery/envs/1935/lib/python3.8/site-packages/docutils/parsers/rst/states.py", line 170, in run
    results = StateMachineWS.run(self, input_lines, input_offset,
  File "/home/docs/checkouts/readthedocs.org/user_builds/astroquery/envs/1935/lib/python3.8/site-packages/docutils/statemachine.py", line 239, in run
    context, next_state, result = self.check_line(
  File "/home/docs/checkouts/readthedocs.org/user_builds/astroquery/envs/1935/lib/python3.8/site-packages/docutils/statemachine.py", line 451, in check_line
    return method(match, context, next_state)
  File "/home/docs/checkouts/readthedocs.org/user_builds/astroquery/envs/1935/lib/python3.8/site-packages/docutils/parsers/rst/states.py", line 3008, in text
    self.section(title.lstrip(), source, style, lineno + 1, messages)
  File "/home/docs/checkouts/readthedocs.org/user_builds/astroquery/envs/1935/lib/python3.8/site-packages/docutils/parsers/rst/states.py", line 327, in section
    self.new_subsection(title, lineno, messages)
  File "/home/docs/checkouts/readthedocs.org/user_builds/astroquery/envs/1935/lib/python3.8/site-packages/docutils/parsers/rst/states.py", line 393, in new_subsection
    newabsoffset = self.nested_parse(
  File "/home/docs/checkouts/readthedocs.org/user_builds/astroquery/envs/1935/lib/python3.8/site-packages/docutils/parsers/rst/states.py", line 281, in nested_parse
    state_machine.run(block, input_offset, memo=self.memo,
  File "/home/docs/checkouts/readthedocs.org/user_builds/astroquery/envs/1935/lib/python3.8/site-packages/docutils/parsers/rst/states.py", line 196, in run
    results = StateMachineWS.run(self, input_lines, input_offset)
  File "/home/docs/checkouts/readthedocs.org/user_builds/astroquery/envs/1935/lib/python3.8/site-packages/docutils/statemachine.py", line 239, in run
    context, next_state, result = self.check_line(
  File "/home/docs/checkouts/readthedocs.org/user_builds/astroquery/envs/1935/lib/python3.8/site-packages/docutils/statemachine.py", line 451, in check_line
    return method(match, context, next_state)
  File "/home/docs/checkouts/readthedocs.org/user_builds/astroquery/envs/1935/lib/python3.8/site-packages/docutils/parsers/rst/states.py", line 2769, in underline
    self.section(title, source, style, lineno - 1, messages)
  File "/home/docs/checkouts/readthedocs.org/user_builds/astroquery/envs/1935/lib/python3.8/site-packages/docutils/parsers/rst/states.py", line 327, in section
    self.new_subsection(title, lineno, messages)
  File "/home/docs/checkouts/readthedocs.org/user_builds/astroquery/envs/1935/lib/python3.8/site-packages/docutils/parsers/rst/states.py", line 393, in new_subsection
    newabsoffset = self.nested_parse(
  File "/home/docs/checkouts/readthedocs.org/user_builds/astroquery/envs/1935/lib/python3.8/site-packages/docutils/parsers/rst/states.py", line 281, in nested_parse
    state_machine.run(block, input_offset, memo=self.memo,
  File "/home/docs/checkouts/readthedocs.org/user_builds/astroquery/envs/1935/lib/python3.8/site-packages/docutils/parsers/rst/states.py", line 196, in run
    results = StateMachineWS.run(self, input_lines, input_offset)
  File "/home/docs/checkouts/readthedocs.org/user_builds/astroquery/envs/1935/lib/python3.8/site-packages/docutils/statemachine.py", line 239, in run
    context, next_state, result = self.check_line(
  File "/home/docs/checkouts/readthedocs.org/user_builds/astroquery/envs/1935/lib/python3.8/site-packages/docutils/statemachine.py", line 451, in check_line
    return method(match, context, next_state)
  File "/home/docs/checkouts/readthedocs.org/user_builds/astroquery/envs/1935/lib/python3.8/site-packages/docutils/parsers/rst/states.py", line 2342, in explicit_markup
    nodelist, blank_finish = self.explicit_construct(match)
  File "/home/docs/checkouts/readthedocs.org/user_builds/astroquery/envs/1935/lib/python3.8/site-packages/docutils/parsers/rst/states.py", line 2354, in explicit_construct
    return method(self, expmatch)
  File "/home/docs/checkouts/readthedocs.org/user_builds/astroquery/envs/1935/lib/python3.8/site-packages/docutils/parsers/rst/states.py", line 2096, in directive
    return self.run_directive(
  File "/home/docs/checkouts/readthedocs.org/user_builds/astroquery/envs/1935/lib/python3.8/site-packages/docutils/parsers/rst/states.py", line 2146, in run_directive
    result = directive_instance.run()
  File "/home/docs/checkouts/readthedocs.org/user_builds/astroquery/envs/1935/lib/python3.8/site-packages/pytest_doctestplus/sphinx/doctestplus.py", line 23, in run
    if re.match('win32', self.content[0]):
  File "/home/docs/checkouts/readthedocs.org/user_builds/astroquery/envs/1935/lib/python3.8/site-packages/docutils/statemachine.py", line 1147, in __getitem__
    return self.data[i]
IndexError: list index out of range

Exception occurred:
  File "/home/docs/checkouts/readthedocs.org/user_builds/astroquery/envs/1935/lib/python3.8/site-packages/docutils/statemachine.py", line 1147, in __getitem__
    return self.data[i]
IndexError: list index out of range
The full traceback has been saved in /tmp/sphinx-err-ma4nbt31.log, if you want to report the issue to the developers.
Please also report this if it was a user error, so that a better error message can be provided next time.
A bug report can be filed in the tracker at <https://github.com/sphinx-doc/sphinx/issues>. Thanks!
```

**Comment by bsipocz** [comment]
Apparently, two directives cannot be used at once...

**Pull request merged**
